src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/OsxIPGlobalProperties.cs

```diff
@@ -20,11 +20,26 @@ public unsafe override TcpConnectionInformation[] GetActiveTcpConnections()
             }
 
             TcpConnectionInformation[] connectionInformations = new TcpConnectionInformation[infoCount];
+            int skip = 0;
```

Comment:
Why manage this as skip rather than managing the number of items copied, i.e. have a:
```C#
int nextResultIndex = 0;
```
that you then just increment as part of:
```C#
connectionInformations[nextResultIndex++] = ...;
```
and then you know if you need to resize if nextResultIndex != connectionInformations.Length at the end.  I realize it's similar, but it seems cleaner, and you only need to increment in one place.